acme_diags/run.py

```diff
@@ -199,7 +199,7 @@ def _find_attrs_with_nondefault_values(self, param):
                 continue
             
             if hasattr(new_instance, attr) and \
-                getattr(new_instance, attr) != getattr(param, attr):
+                getattr(new_instance, attr) != getattr(param, attr):   #this only valid when the attri values are lists not numpy array
```

Comment:
Add space after `#`. `attri` should be `attr`. If this is only valid if `attr` values are lists, what happens in the code if they are not?

Reply:
i.g., if the `attr` value is a numpy array, the logic expression returns a Boolean array, which caused the problem.

Reply:
So, by the time the code reaches this point, we know for sure that `attr` values are lists, not numpy arrays?

Reply:
yes. The `check_value` function in the parameter scripts should take care of this.

Reply:
Another way is to find a Boolean expression valid to check match for both list and numpy array.